clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1971,7 +1971,7 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
     if (KernelDecl->hasAttr<SYCLSimdAttr>())
```

Comment:
I suggest removing this and moving the code out of `else `below. I think the only code you need in this handler is -  

```
handleAccessorPropertyList(Params.back(), RecordDecl, Loc);
const auto *AccessorSpecializationDecl =
        cast<ClassTemplateSpecializationDecl>(RecordDecl);
    const TemplateArgument &AccessModeArg =
        AccessorSpecializationDecl->getTemplateArgs().get(2);

 // Add implicit attribute to parameter decl when it is a read only
 // SYCL accessor.
 if (isReadOnlyAccessor(AccessModeArg))
    Params.back()->addAttr(
        SYCLAccessorReadonlyAttr::CreateImplicit(SemaRef.getASTContext()));
 
   Params.back()->addAttr(
          SYCLAccessorPtrAttr::CreateImplicit(SemaRef.getASTContext()));
```

Basically SYCLAccessorPtrAttr will be created for all accessors. SYCLAccessorReadonlyAttr will be created in addition for read only accessors. IMO this is cleaner